src/main/java/github/grace000/erversay/ServerConnection.java

```diff
@@ -0,0 +1,52 @@
+package github.grace000.erversay;
+
+import java.io.*;
+import java.net.*;
+
+
+public class ServerConnection {
+    private ServerSocket serverSocket;
+    private Socket socket;
+    private InputStream inputStream;
+    private OutputStream outputStream;
+
+    public void serveClient(String response) throws IOException {
+        acceptConnection();
+        getInputOutputStreams();
+        writeResponse(response);
+    }
+
+    public void listenForConnection(int port) {
+        try {
+            serverSocket = new ServerSocket(port);
+            System.out.println("Server listening ...");
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void acceptConnection() {
+        try {
+            socket = serverSocket.accept();
+            System.out.println("New client connected");
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    private void getInputOutputStreams() {
```

Comment:
Totally legit name, unfortunately it kind of indicates that it's returning a property called inputOutputStreams, "receive" might be a good synonym here?

Reply:
+1 and also I wonder if it might not be worth separating the input and output stream.